### PR TITLE
GCI-951: Implement authorisation

### PR DESCRIPTION
**Pull request by VaughanJackson123**
* Implemented authorisation for the Orders API, according to the requirements captured in the Authorisation column of the two tables on this page: https://companieshouse.atlassian.net/wiki/spaces/CHS/pages/1515684399/GCI-926+Investigate+authentication+and+authorisation+checks+in+Orders+API.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/orders/api/controller/OrderController.java [hunk 1]
```diff
@@ -25,15 +25,18 @@
 
     private final OrderService orderService;
 
+    public static final String ORDER_ID_PATH_VARIABLE = "id";
+
+    /** <code>${uk.gov.companieshouse.orders.api.orders}/{id}</code> */
```
Comment (nabeelamirch) [inline src/main/java/uk/gov/companieshouse/orders/api/controller/OrderController.java:30]: Wow, what's this?

Comment (VaughanJackson123) [reply src/main/java/uk/gov/companieshouse/orders/api/controller/OrderController.java:30]: It's a Javadoc comment that attempts to restore some of the clarity lost from introducing the extra path variable constant.

File: src/test/java/uk/gov/companieshouse/orders/api/controller/BasketControllerIntegrationTest.java [hunk 2]
```diff
@@ -458,13 +460,14 @@ public void patchBasketPaymentDetailsReturnsOK() throws Exception {
                 .header(REQUEST_ID_HEADER_NAME, TOKEN_REQUEST_ID_VALUE)
                 .header(ERIC_IDENTITY_TYPE_HEADER_NAME, ERIC_IDENTITY_API_KEY_TYPE_VALUE)
                 .header(ERIC_IDENTITY_HEADER_NAME, ERIC_IDENTITY_VALUE)
```
Comment (nabeelamirch) [inline src/test/java/uk/gov/companieshouse/orders/api/controller/BasketControllerIntegrationTest.java:462]: Does  uk.gov.companieshouse.api.util.security.EricConstants have any of the other constants that we can use, like ERIC_IDENTITY_HEADER_NAME? Which we currently have in the TestConstants file.

Comment (VaughanJackson123) [reply src/test/java/uk/gov/companieshouse/orders/api/controller/BasketControllerIntegrationTest.java:462]: It has some, but not all of the values we use. I think if it had them all, I would have shifted completely over to using it. However, given its incompleteness I have not done this.

**Pull request merged**